src/main/java/bio/terra/service/duos/DuosDao.java

```diff
@@ -0,0 +1,104 @@
+package bio.terra.service.duos;
+
+import bio.terra.model.DuosFirecloudGroupModel;
+import com.google.common.annotations.VisibleForTesting;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class DuosDao {
+  private static final Logger logger = LoggerFactory.getLogger(DuosDao.class);
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private final String tdrServiceAccountEmail;
+
+  @Autowired
+  public DuosDao(
+      NamedParameterJdbcTemplate jdbcTemplate,
+      @Qualifier("tdrServiceAccountEmail") String tdrServiceAccountEmail) {
+    this.jdbcTemplate = jdbcTemplate;
+    this.tdrServiceAccountEmail = tdrServiceAccountEmail;
+  }
+
+  @VisibleForTesting
+  String getTdrServiceAccountEmail() {
+    return tdrServiceAccountEmail;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public boolean insertFirecloudGroup(
+      String duosId, String firecloudGroupName, String firecloudGroupEmail) {
+    String sql =
+        """
+        INSERT INTO duos_firecloud_group
+        (duos_id, firecloud_group_name, firecloud_group_email, created_by) VALUES
+        (:duos_id, :firecloud_group_name, :firecloud_group_email, :created_by)
+        """;
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("duos_id", duosId)
+            .addValue("firecloud_group_name", firecloudGroupName)
+            .addValue("firecloud_group_email", firecloudGroupEmail)
+            .addValue("created_by", tdrServiceAccountEmail);
+    int rowsAffected = jdbcTemplate.update(sql, params);
+    boolean insertSucceeded = (rowsAffected == 1);
```

Comment:
Is there a way that this can be `false`; that is, what could cause `rowsAffected` to be `0` but no exception to be thrown?

Reply:
I see, if `duosId` is not found in the database, no exception is thrown and `update()` returns `0`.

Reply:
Actually, I think your initial question is right on.  If we haven't inserted a row, then something exceptional occurred and would have shown up as a thrown exception: violation of a constraint, etc.  We're better served by allowing the exception to bubble up rather than obfuscating it.

Reply:
Making this change ended up with the added benefit of streamlining error handling at the caller, so thanks!